application/core/LSYii_Application.php

```diff
@@ -515,4 +518,52 @@ public function getTwigCustomExtensionsConfig($sUsertwigextensionrootdir, $aAppl
 
         return $aApplicationConfig;
     }
+
+    /**
+     * @inheritdoc
+     * Special handling for SEO friendly URLs
+        */
+    public function createController($route, $owner=null)
+    {
+        $controller = parent::createController($route, $owner);
+
+        // If no controller is found by standard ways, check if the route matches
+        // an existing survey's alias.
+        if (is_null($controller)) {
+            $controller = $this->createControllerFromShortUrl($route);
+        }
+
+        return $controller;
+    }
+
+    /**
+     * Create controller from short url if the route matches a survey alias.
+     * @param string $route the route of the request.
+     * @return array<mixed>|null
+     */
+    private function createControllerFromShortUrl($route)
+    {
+        // We can only determine Survey aliases if the column exists (we can't asume it does, because db update runs from the controller).
+        if ($this->getDb()->getSchema()->getTable('{{surveys_languagesettings}}')->getColumn('surveyls_alias')) {
+            $alias = explode("/", $route)[0];
+            $languageSettings = SurveyLanguageSetting::model()->with([
+                'survey' => [
+                    'select' => false,
+                    'joinType' => 'INNER JOIN',
+                ],
+            ])->findByAttributes(['surveyls_alias' => $alias]);
+            if (!empty($languageSettings)) {
+                // Override language
+                if (isset($_GET['lang'])) {
+                    $_GET['lang'] = $languageSettings->surveyls_language;
+                }
+                if (isset($_POST['lang'])) {
+                    $_POST['lang'] = $languageSettings->surveyls_language;
+                }
+                return $this->createController("survey/index/sid/" . $languageSettings->surveyls_survey_id);
```

Comment:
> I'm more for `$this->createController`

Please take a look at the signature of the method.
This is not a URL creation method, but a controller creation.
Params are not received.

Still, we will review it as we think we should call parent::createController, mainly to avoid recursion.

Reply:
> /lilurlfr : goes to survey in fr
> but
> /lilurlfr?lang=en : goes to survey in en

We discussed and thought about this (I admit, in the past).
- Each language can handle a different alias
- Still, they can use the same alias
- When looking for a survey from an alias, if there is a language in the querystring, we use it as criteria to find the survey and the 
- So, your prior case would work if both languages share the same alias.
- If EN doesn't have `lilurlfr` as an alias, survey will not be found.

An enhancement we can do is:
- When searching a survey by alias, don't filter by language, any language settning will be OK.
- At last, we just wan't to know the survey id.